### PR TITLE
Better dynamics

### PR DESCRIPTION
**Pull request by timspainNERSC**
# Better dynamics
## Fixes \#388

---
# Change Description

Improve the connection between the main model and the dynamics
- Clean out the fake forcing code
- Pass ice velocity components to and from the dynamics
- Clean up the smesh files
- Outright delete commented code

### CODE REVIEW

**Comment by einola** [comment]
Thomas should put his bug fix in this branch.

**Comment by einola** [comment]
> Thomas should put his bug fix in this branch.

This refers to commit [18fe081](https://github.com/nextsimhub/nextsimdg/pull/389/commits/18fd0816b58ab917b9814e4d6ccae446c5ce760e).

File: dynamics/src/ParametricMap.cpp [hunk 1]
```diff
@@ -44,7 +44,8 @@ namespace Nextsim
 	if (smesh.CoordinateSystem == SPHERICAL)
 	  {
 	    const Eigen::Matrix<Nextsim::FloatType, 1, GAUSSPOINTS(DG)> cos_lat = (ParametricTools::getGaussPointsInElement<GAUSSPOINTS1D(DG)>(smesh, eid).row(1).array()).cos();
-	    AdvectionCellTermY[eid] = PSIy<DG, GAUSSPOINTS1D(DG)>.array().rowwise() * (dxT.row(0).array() * cos_lat.array()) - PSIx<DG, GAUSSPOINTS1D(DG)>.array().rowwise() * (dyT.row(0).array() * cos_lat.array());
+	    //	    AdvectionCellTermY[eid] = PSIy<DG, GAUSSPOINTS1D(DG)>.array().rowwise() * (dxT.row(0).array() * cos_lat.array()) - PSIx<DG, GAUSSPOINTS1D(DG)>.array().rowwise() * (dyT.row(0).array() * cos_lat.array());
```
Comment (MarionBWeinzierl) [inline dynamics/src/ParametricMap.cpp:47]: can the commented-out code be removed?

**Review by MarionBWeinzierl — APPROVED** [review]
I don't understand much of the Physics changes in this, but code-wise it looks sensible

**Review by MarionBWeinzierl — APPROVED** [review]

**Pull request merged**